love/src/components/Facility/Map/Device.jsx

```diff
@@ -0,0 +1,366 @@
+  }
+
+  componentDidUpdate() {}
+
+  renderStates(ctx) {
+    switch (this.props.states[ctx]) {
+      case true:
+        return `${styles.statusOk}`;
+      case false:
+        return `${styles.statusAlert}`;
+      default:
+        return `${styles.statusNull}`;
+    }
+  }
+
+  statusToStyle(ctx) {
+    switch (ctx) {
+      case 'true':
+        return `${styles.bgStatusOk}`;
+      case 'false':
+        return `${styles.bgStatusAlert}`;
+      default:
+        return `${styles.bgStatusNull}`;
+    }
+  }
+
+  statusTextToStyle(ctx) {
+    switch (ctx) {
+      case 'true':
+        return `${styles.statusOk}`;
+      case 'false':
+        return `${styles.statusAlert}`;
+      default:
+        return `${styles.statusDisabled}`;
+    }
+  }
+
+  renderArrow() {
+    return (
+      <g
+        id={this.id}
+        transform={'translate(' + (this.props.width - 14) + ' 4)'}
+        pointerEvents="all"
+        onMouseUp={(e) => this.show()}
+      >
+        <rect className={styles.arrowBg} width={10} height={10} />
+        <polyline
+          className={styles.arrow}
+          points=".35 .35 4.35 4.35 8.35 .35"
+          transform={'translate(0.5 3) rotate(' + (this.state.hidden ? -90 : 0) + ')'}
+          transform-origin="4 2"
+        />
+      </g>
+    );
+  }
+
+  // This function renders the badge style alarm norifications on devices.
+  renderAlarms() {
+    const height = this.props.height;
+    const width = this.props.width;
+    const hidden = this.state.hidden;
+    const alarms = this.props.alarms;
+    const alarmsKeys = Object.keys(alarms);
+    const rowLength = Math.round(width / 42 - 1);
+
+    let transX;
+    let transY;
+    let currentRow;
+    let currentCol;
+
+    return alarmsKeys.map((x, k) => {
+      if (alarms[x].state) {
+        transX = 0;
+        transY = 0;
+        currentRow = Math.ceil((k + 1) / rowLength);
+        currentCol = (k + 1) % currentRow;
+
+        console.log(
+          alarms[x].name +
+            ' row length:' +
+            rowLength +
+            ' current row:' +
+            currentRow +
+            ' current col:' +
+            currentCol +
+            ' k:' +
+            k,
+        );
+
+        transX = 42 * (k + 1 - rowLength * (currentRow - 1)) - 42;
+
+        if (hidden) {
+          transY = 12 * (currentRow - 1);
+        } else {
+          transY = 12 * (currentRow - 1) + height;
+        }
+
+        return (
+          <g className={styles.alarm} transform={'translate(' + transX + ' ' + transY + ')'}>
+            <rect className={styles.bgStatusAlert} width={40} height={10} rx={5} ry={5} transform={'translate(0 33)'} />
+            <text className={styles.statusAlert} transform={'translate(20 39)'} textAnchor="middle">
+              <tspan>{alarms[x].name}</tspan>
+            </text>
+          </g>
+        );
+      }
+    });
+  }
+
+  //This function renders the Decive parameters. This is the detailed information that is collapsible
+
```

Comment:
This line should be removed (we have to check why the linter and formatter didn't remove this)